pom.xml

```diff
@@ -621,7 +621,7 @@
         <concurrent.version>1.3.4</concurrent.version>
         <xbean-spring.version>4.21</xbean-spring.version>
         <netty.version>4.1.101.Final</netty.version>
-        <cucumber.version>7.14.1</cucumber.version>
+        <cucumber.version>7.15.0</cucumber.version>
```

Comment:
seems like a fixup!  JAMES-3691 Update Cucumber to 7.14.1 

i wouldn't mind a longer explanation of why rrt tests needed to be dropped

Reply:
> i wouldn't mind a longer explanation of why rrt tests needed to be dropped

Do you mean the data-file RRT test removal? (https://github.com/apache/james-project/pull/1963/commits/ccce958046ec774097b6eb73329a48de97c62e92#diff-c7eba70ed09589fac26f92d4756f4e266aa408f64337f17131f77f1b1333bb13)
![image](https://github.com/apache/james-project/assets/55171818/fce51ebe-2079-498a-a116-623ae8a2f4cf)

cc @chibenwa any rationale behind that?

Reply:
the commit comment only says a cryptic 
> RRT cucumber tests do not account for read-only 
what seems weird is that these tests have been aroun for a while and I don't understand why moving to jdk21 would affect them. 
Does it mean that moving to jdk21 affects : 
- the capabilities of the data-file rrt ? 
- only the tests and we decided to just drop the tests (seems weird)
- was this already broken somehow before the upgrade and is actually a cleanup commit that's unrelated to jdk21

Reply:
> only the tests and we decided to just drop the tests (seems weird)

It seems to fall in this case.

The test was still fine before this work.

RRT tests for data-file is just only 1 test (with `readonly` tag): https://github.com/apache/james-project/blob/master/server/data/data-library/src/test/resources/cucumber/rewrite_tables.feature#L4 though.

I will try to see if we can still keep that, otherwise only 1 test lost is minimal and could be acceptable IMO.